components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credential-details/node-credential-details.component.ts

```diff
@@ -0,0 +1,188 @@
+    private fb: FormBuilder,
+    private router: Router,
+    private layoutFacade: LayoutFacadeService
+  ) {
+    this.sshForms = this.fb.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required],
+      key: ['', Validators.required]
+    });
+    this.winrmForms = this.fb.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required]
+    });
+    this.sudoForms = this.fb.group({
+      password: ['', Validators.required],
+      options: ['', Validators.required]
+    });
+
+    this.updateForm = this.fb.group({
+      id: [''],
+      name: ['', Validators.required],
+      type: ['', Validators.required],
+      ssh: this.sshForms,
+      winrm: this.winrmForms,
+      sudo: this.sudoForms
+    });
+
+    this.resetForm = this.fb.group({
+      id: [''],
+      name: ['', Validators.required],
+      type: ['', Validators.required],
+      ssh: this.sshForms,
+      winrm: this.winrmForms,
+      sudo: this.sudoForms
+    });
+    this.getNodeCred();
+  }
+
+  ngOnInit() {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+    this.store.select(routeURL).pipe(takeUntil(this.isDestroyed))
+      .subscribe((url: string) => {
+        this.url = url;
+        const [, fragment] = url.split('#');
+        this.tabValue = (fragment === 'reset') ? 'reset' : 'details';
+      });
+
+    this.store.select(updateStatus).pipe(
+      takeUntil(this.isDestroyed),
+      filter(state => this.saveInProgress || this.resetInProgress && !pending(state)))
+      .subscribe((state) => {
+        if (this.tabValue === 'details') {
+          this.saveInProgress = false;
+          this.saveSuccessful = (state === EntityStatus.loadingSuccess);
+          this.updateForm.markAsPristine();
+        } else {
+          this.resetInProgress = false;
+          this.resetSuccessful = (state === EntityStatus.loadingSuccess);
+          this.resetForm.markAsPristine();
+        }
+        this.getNodeCred();
+      });
+  }
+
+
+  onSelectedTab(event: { target: { value: NodeCredentialTabName } }) {
+    this.tabValue = event.target.value;
+    this.router.navigate([this.url.split('#')[0]], { fragment: event.target.value });
+  }
+
+  selectChangeHandlers(id: string): void {
+    this.passwordSelected = id;
+  }
+
+  saveNodeCredential(data: NodeObject): void {
+    let credBody;
+    if (this.tabValue === 'details') {
+      this.saveSuccessful = false;
+      this.saveInProgress = true;
+      credBody = {
+        id: data.id,
+        name: data.name,
+        type: this.nodeCredential.type,
+        data: this.nodeCredential.data,
+        tags: this.nodeCredential.tags
+      };
+    } else {
+      this.resetSuccessful = false;
+      this.resetInProgress = true;
+      credBody = this.saveCred.getNodeCredentialCreate(data);
+    }
+
+    this.store.dispatch(new UpdateNodeCredential(credBody));
+    this.resetForm.reset();
+  }
+
+  getNodeCred() {
```

Comment:
1. add return type
2. Would suggest renaming to, e.g. "setupNodeSubscriptions" to make it clear that it is an initialization function